parlai/torchscript/modules.py

```diff
@@ -817,44 +1013,161 @@ def gpt2_tokenize(self, text: str):
         """
         Tokenize using Gpt2 BPE tokenizer.
         """
-        return self.bpe_tokenize(text)
-
-    def tokenize(self, text: str) -> List[str]:
-        """
-        Return a sequence of tokens from the iterable.
-
-        Also handles special tokens for some tokenizers
-        """
-
-        # calls the selected tokenizer function e.g. 're' => re_tokenize(text)
-        word_tokens = self.gpt2_tokenize(text)
-
-        return word_tokens
+        return self.gpt2_bpe.encode(text)
 
     def bpe_tokenize(self, text: str) -> List[str]:
         """
         Return a sequence of BPE-tokens from the text.
         """
-        return self.bpe.encode(text)
+        return self.subword_bpe.encode(text)
 
-    def txt2vec(self, text: str) -> List[int]:
+    def txt2vec(self, text: str, dict_tokenizer: str) -> List[int]:
         """
         Convert a string to a vector (list of ints).
 
         First runs a sentence tokenizer, then a word tokenizer.
         """
         itr: List[int] = []
-        for token in self.tokenize(str(text)):
-            itr.append(self._word_lookup(token))
+        if dict_tokenizer == 'gpt2' or dict_tokenizer == 'slow_bytelevel_bpe':
+            for token in self.gpt2_tokenize(str(text)):
+                itr.append(self._word_lookup(token))
+        elif dict_tokenizer == 'bpe':
+            for token in self.bpe_tokenize(str(text)):
+                itr.append(self._word_lookup(token))
+        else:
+            raise NotImplementedError
         return itr
 
-    def vec2txt(self, vector: List[int]) -> str:
+    def vec2txt(self, vector: List[int], dict_tokenizer: str) -> str:
         """
         Convert a vector of IDs to a string.
 
         Converts a vector (iterable of ints) into a string, with each token separated by
         the delimiter (default ``' '``).
         """
         tokens = [self._index_lookup(idx) for idx in vector]
-        text = self.bpe.decode(tokens)
+        if dict_tokenizer == 'gpt2' or dict_tokenizer == 'slow_bytelevel_bpe':
+            text = self.gpt2_bpe.decode(tokens)
+        elif dict_tokenizer == 'bpe':
+            text = self.subword_bpe.decode(tokens)
+        else:
+            raise NotImplementedError
         return text
+
+
+class TorchScriptTransformerClassifier(nn.Module):
```

Comment:
This file is getting huge. I'd suggest we break modules.py up.

Reply:
I moved tokenizer out of module.py. Do we want to split the modules further? e.g., generator_module and classifier_module.

Reply:
I'll let you decide

Reply:
I think it's good for now. If there are more models to add. We can group them further.